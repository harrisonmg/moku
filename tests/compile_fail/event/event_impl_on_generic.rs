use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    use machine::BlinkyState;

    struct Event<T>;

    impl<T> StateMachineEvent for Event<T> {}

    struct Top;

    impl TopState<BlinkyState, Event> for Top {}
}

fn main() {}
