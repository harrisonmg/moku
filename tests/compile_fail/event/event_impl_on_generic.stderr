error: `moku::StateMachineEvent` must be implemented on a plain enum or struct
  --> tests/compile_fail/event/event_impl_on_generic.rs:14:35
   |
14 |     impl<T> StateMachineEvent for Event<T> {}
   |                                   ^^^^^
